### PR TITLE
feat: Added core Concepts of AndroidNavigation via Jetpack 

### LINKED ISSUE #17: Add documentation for AndroidNavigationComponents
Fixes #16

### PR DESCRIPTION
**Pull request by Annysah**
Fixes #17 
Added docs for Android Navigation Component Library

### CODE REVIEW

**Comment by s-ayush2903** [comment]
What happened @Annysah ??

**Comment by raghavSharma1472** [comment]
Please semantify your PR title within appropriate qualifier like _feat_, _docs_

**Comment by Annysah** [comment]
@s-ayush2903 I was trying to figure out why my PR did not pass all the checks

**Comment by Annysah** [comment]
@raghavSharma1472 how can I do that?

**Comment by s-ayush2903** [comment]
Ah, no need to worry about it, that is just a semantic check which we are using so that devs make semantic PR titles and it may become easier for us to check a PR if it adds some `feature` or `docs`  or any other technical stuff. You may change your PR title by clicking _Edit_ right adjacent to your PR title
|Edit PR Title|
|----|
|editing|

**Comment by Annysah** [comment]
Thanks @s-ayush2903 , I've been able to make the corrections

**Comment by s-ayush2903** [comment]
@Annysah I would request you to create an Issue corresponding to this PR about adding documentation for _AndroidNavigationComponents_ and then link that issue with this PR. As the issue on which you were assigned is a very big one and has a lot of topics to cover, so it would be better if we have many child issues related to the parent one and we can close 'em on the go with the PRs corresponding to 'em.

PS: You may link an issue to a PR via adding: `Fixes #issueNumber` in the PR's description

**Comment by Annysah** [comment]
Okay, I will do that.

File: docs/library_name.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Every NavHost possess its own corresponding NavController(an object which helps in navigating to destinations within a NavHost in apps) that can be used in your Kotlin or Java code.
+
+## How to Implement Navigation Component In Your Projects
+- To use the Navigation component, it requires the Android 3.3 or higher and is dependent on Java 8 language features. Note: You will need to include Navigation support in your project by adding dependencies to your app's `build.gradle` file, [Click here to learn more](https://developer.android.com/topic/libraries/architecture/adding-components#navigation).
+
+- Add a navigation graph in your project. To do this;
+ 1. In the Project window, right click on the `res` directory
+ 2. Then click New > Android Resource File.
+ 3. Type a name in the File name field e.g "nav_graph".
```
Comment (s-ayush2903) [inline docs/library_name.md:28]:
```suggestion
 3. Type a name in the File name field e.g `nav_graph`.
```
This'll look better

File: docs/library_name.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ 4. From the resource type drop-down list, click on Navigation then click OK to proceed.
+
+- After which it opens in the navigation editor to edit the navigation graph visually or edit the XML directly.
+
+- Add a NavHostFragment via the Layer Editor;
+ 1. In the [Layout Editor](https://developer.android.com/studio/write/layout-editor) double click on your activity's layout XML file in your list of project files.  
+ 2. Search for "NavHostFragment".
```
Comment (s-ayush2903) [inline docs/library_name.md:35]: You should replace inverted commas with backticks(``)

File: docs/library_name.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+- Add a navigation graph in your project. To do this;
+ 1. In the Project window, right click on the `res` directory
+ 2. Then click New > Android Resource File.
+ 3. Type a name in the File name field e.g "nav_graph".
+ 4. From the resource type drop-down list, click on Navigation then click OK to proceed.
+
+- After which it opens in the navigation editor to edit the navigation graph visually or edit the XML directly.
+
+- Add a NavHostFragment via the Layer Editor;
```
Comment (s-ayush2903) [inline docs/library_name.md:33]: The semi-colon here is not needed.

**Comment by s-ayush2903** [comment]
* Ah, you did the opposite, linking the PR with the issue, no worries
* Issues in the repos mean the problems that are in the module or the changes that are needed, and the PullRequest(PR) consists of the code which on being merged solves the issue, this is the open-source workflow : ) 
* You are supposed to _link Issue to the PR_ doing this will lead to closure of the linked issue as soon as the corresponding Pull Request gets merged. For example take a look at my [this](https://github.com/njaLocer/Locer-Android-Client/pull/8) PR.

File: docs/library_name.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+## Android Navigation Components
+
+Navigation components is simply a collection of libraries, plugin and tooling for making Android Navigation easier. Asides making Android Navigation easier, it also helps in handling backstacks, automating fragment transactions, handling transition animations and simplifying deep linking.
```
[the remark this answers is not in the document]
Comment (Annysah) [reply docs/library_name.md:3]: Can I edit it directly on here?

[the remark this answers is not in the document]
Comment (s-ayush2903) [reply docs/library_name.md:3]: Yeah, definitely from your fork(at github) by checking out to the branch from which you opened the PR

**Comment by s-ayush2903** [comment]
Just change the file name to a more appropriate one and we'll be merging this, rest looks fine!

**Review by s-ayush2903 — APPROVED** [review]
No worries, we can merge it right here, will change the file name in future PRs

**Comment by s-ayush2903** [comment]
Good Work and thanks for incorporating suggested changes :tada:

**Comment by Annysah** [comment]
Thank you so much @s-ayush2903

**Comment by s-ayush2903** [comment]
No worries : )

**Comment by Annysah** [comment]
@s-ayush2903 Can I work on the documentation of the other libraries mentioned in the issue I worked on previously?

**Comment by s-ayush2903** [comment]
Yeah sure, definitely, create a mini-issue corresponding to the topic you are interested in covering and claim it from there via commenting.

**Comment by Annysah** [comment]
Please how can I create a mini-issue?

**Comment by s-ayush2903** [comment]
Nothing special, by mini-issue meant creating an issue on one of the topics mentioned in [this comment](https://github.com/njaLocer/Locer-Android-Client/issues/11#issuecomment-701609879), sorry for the inconvenience, I used the incorrect word maybe

**Comment by Annysah** [comment]
No problem @s-ayush2903

**Review by raghavSharma1472 — COMMENTED** [review]
Nice Work, Keep it up!

**Pull request merged**